src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/incidents/QuestionnaireResource.kt

```diff
@@ -0,0 +1,198 @@
+}
+
+@Schema(description = "Questionnaire")
+@JsonInclude(JsonInclude.Include.NON_NULL)
+data class QuestionnaireResponse(
+  @Schema(description = "The unique identifier of the questionnaire")
+  val id: Long,
+  @Schema(description = "A description of the questionnaire", example = "Escape from Establishment")
+  val description: String?,
+  @Schema(description = "Code to identify this questionnaire", example = "ESCAPE_EST")
+  val code: String,
+  @Schema(description = "If the questionnaire is active", example = "true")
+  val active: Boolean,
+  @Schema(description = "Sequence value of the questionnaires", example = "1")
+  val listSequence: Int,
+  @Schema(description = "List of Questions (and associated Answers) for this Questionnaire")
+  val questions: List<QuestionResponse> = listOf(),
+  @Schema(description = "The date the questionnaire is no longer used")
+  val expiryDate: LocalDate? = null,
+  @Schema(description = "Questionnaire created date", required = true)
+  val createdDate: LocalDateTime,
+  @Schema(description = "Questionnaire created by", required = true)
+  val createdBy: String,
+  @Schema(description = "Questionnaire modified date")
+  val modifiedDate: LocalDateTime? = null,
+  @Schema(description = "Questionnaire modified by")
+  val modifiedBy: String? = null,
+)
+
+@JsonInclude(JsonInclude.Include.NON_NULL)
+data class QuestionResponse(
+  @Schema(description = "The question id", required = true)
+  val id: Long,
+  @Schema(description = "The question text", required = true)
+  val question: String,
+  @Schema(description = "List of Answers to this question")
+  val answers: List<AnswerResponse> = listOf(),
+  @Schema(description = "If the question is active", example = "true")
+  val active: Boolean,
+  @Schema(description = "The date the question is no longer used")
+  val expiryDate: LocalDate? = null,
+  @Schema(description = "If the question has multiple answers", example = "true")
+  val multipleAnswers: Boolean,
+  @Schema(description = "The question id used to set the listSequence values", example = "1")
+  val questionSequence: Int,
+  @Schema(description = "The order of the questions", example = "1")
+  val listSequence: Int,
+)
+
+@JsonInclude(JsonInclude.Include.NON_NULL)
+data class AnswerResponse(
+  @Schema(description = "The answer id", required = true)
+  val id: Long,
+  @Schema(description = "The answer text", required = true)
+  val answer: String,
+  @Schema(description = "The answer id used to set the listSequence values", example = "1")
+  val answerSequence: Int,
+  @Schema(description = "The order of the answers", example = "1")
+  val listSequence: Int,
+  @Schema(description = "If the answer is active", example = "true")
+  val active: Boolean,
+  @Schema(description = "The date the answer is no longer used")
+  val expiryDate: LocalDate? = null,
+  @Schema(description = "Question to be asked following this answer")
+  var nextQuestion: QuestionResponse? = null,
```

Comment:
Could we get a recursive stack over flow here? I guess the service needs some logic so it only goes one level deep